storage_controller/src/service.rs

```diff
@@ -3764,10 +3811,169 @@ impl Service {
 
         Ok(TimelineCreateResponseStorcon {
             timeline_info,
-            safekeepers,
+            safekeepers: selected_safekeepers,
         })
     }
 
+    pub(crate) async fn put_timeline_import_status(
+        self: &Arc<Self>,
+        req: PutTimelineImportStatusRequest,
+    ) -> Result<(), ApiError> {
+        let res = self
+            .persistence
+            .update_timeline_import(req.tenant_shard_id, req.timeline_id, req.status)
+            .await;
+        let timeline_import = match res {
+            Ok(Ok(Some(timeline_import))) => timeline_import,
+            Ok(Ok(None)) => {
+                return Ok(());
+            }
+            Ok(Err(logical_err)) => {
+                return Err(logical_err.into());
+            }
+            Err(db_err) => {
+                return Err(db_err.into());
+            }
+        };
```

Comment:
```
Ok(Ok(None))
```

Nobody will know what that means without navigating to the return type of `update_timeline_import`

Can we improve readability in some way?

Reply:
Guess I could return something more typed, but then I'd deviate from the `DatabaseResult` standard interface.
Added a comment https://github.com/neondatabase/neon/pull/11345/commits/77d37c9d01fbcd507ce57333a12ffeb158fea3f2